index.js

```diff
@@ -23,7 +23,7 @@ function Argv (processArgs, cwd) {
         .map(function (x) {
             // ignore the node bin, specify this in your
             // bin file with #!/usr/bin/env node
-            if (~x.indexOf('node') || ~x.indexOf('iojs')) return;
+            if (/^(node|iojs)\b/.test(x)) return;
```

Comment:
Done.  I also made this check only apply to `process.argv[0]` - it's possible, though pretty unlikely, that someone could name their binary `/path/to/project/scripts/node`.